contracts/test/LightClientV2.sol

```diff
@@ -56,6 +56,22 @@ contract LightClientV2 is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     /// @notice mapping to store light client states in order to simplify upgrades
     mapping(uint32 index => LightClientState value) public states;
 
+    /// @notice the address of the prover that can call the newFinalizedState function when the
+    /// contract is
+    /// in permissioned prover mode. This address is address(0) when the contract is not in the
+    /// permissioned prover mode
+    address public permissionedProver;
+
+    /// @notice a flag that indicates when a permissioned provrer is needed
+    bool public permissionedProverEnabled;
+
+    /// @notice an array to store the L1 Block Heights where the finalizedState was updated
+    uint256[] public stateUpdateBlockNumbers;
```

Comment:
Would it be better to have this as `uint128[]`? I'm not 100% sure how solidity lays out memory under the hood but we could potentially half the size of every element in this array if we are okay with the max value of an entry being (2^128) -1. 
However, I also recognize this is a test file and gas costs might not be a factor here.

Reply:
actually looking at this again, since we're unsure about the future block sizes and this stores l1 blocks, sticking with `uint256[]` might be best.